service/controller/appcatalog/v1/resource/index/current_test.go

```diff
@@ -122,13 +122,15 @@ func Test_Resource_GetCurrentState(t *testing.T) {
 				t.Fatal("expected non-nil ConfigMap got nil")
 			}
 
-			cm, err := toConfigMap(result)
-			if err != nil {
-				t.Fatalf("error == %#v, want nil", err)
-			}
+			if result != nil {
```

Comment:
if not check this condition, then we are falling into memory error.

Reply:
I think we should check if `len(result) == 1`.

Reply:
is it okay when `result` is nil?

Reply:
Yes.

Reply:
https://play.golang.org/p/X9R3OmCAxcm